.changeset/clear-olives-know.md

```diff
@@ -0,0 +1,177 @@
+---
+"thirdweb": minor
+---
+
+Adds a new `Bridge` module to the thirdweb SDK to access the Universal Bridge.
+
+## Features
+
+### Buy & Sell Operations
+
+The Bridge module makes it easy to buy and sell tokens across chains:
+
+- `Bridge.Buy` - For specifying the destination amount you want to receive
+- `Bridge.Sell` - For specifying the origin amount you want to send
+
+Each operation provides two functions:
+1. `quote` - Get an estimate without connecting a wallet
+2. `prepare` - Get a finalized quote with transaction data
+
+#### Buy Example
+
+```typescript
+import { Bridge, toWei } from "thirdweb";
+
+// First, get a quote to see approximately how much you'll pay
+const buyQuote = await Bridge.Buy.quote({
+  originChainId: 1, // Ethereum
+  originTokenAddress: "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE", // ETH
+  destinationChainId: 10, // Optimism
+  destinationTokenAddress: "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE", // ETH
+  buyAmountWei: toWei("0.01"), // I want to receive 0.01 ETH on Optimism
+  client: thirdwebClient,
+});
+
+console.log(`To get ${buyQuote.destinationAmount} wei on destination chain, you need to pay ${buyQuote.originAmount} wei`);
+
+// When ready to execute, prepare the transaction
+const preparedBuy = await Bridge.Buy.prepare({
+  originChainId: 1,
+  originTokenAddress: "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE",
+  destinationChainId: 10,
+  destinationTokenAddress: "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE",
+  buyAmountWei: toWei("0.01"),
+  sender: "0x...", // Your wallet address
+  receiver: "0x...", // Recipient address (can be the same as sender)
+  client: thirdwebClient,
+});
+
+// The prepared quote contains the transactions you need to execute
+console.log(`Transactions to execute: ${preparedBuy.transactions.length}`);
+```
+
+#### Sell Example
+
+```typescript
+import { Bridge, toWei } from "thirdweb";
+
+// First, get a quote to see approximately how much you'll receive
+const sellQuote = await Bridge.Sell.quote({
+  originChainId: 1, // Ethereum
+  originTokenAddress: "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE", // ETH
+  destinationChainId: 10, // Optimism
+  destinationTokenAddress: "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE", // ETH
+  sellAmountWei: toWei("0.01"), // I want to sell 0.01 ETH from Ethereum
```

Comment:
This also makes the dev handle the decimals themselves, can be tricky.

Wonder if we should handle the conversion?

Reply:
maybe we can make a separate conversion helper that handles it, passing amount + contract, but leave this API pure 🤔

Reply:
Yeah, interstate accepted both which I thought was a little messy. I figured bigint/wei would be best since usually apps are storing onchain values as bigints anyway but we could definitely give a conversion helper. (/routes does return token decimals btw)